### PR TITLE
FEAT: RESTful endpoint - Battle

### PR DESCRIPTION
**Pull request by naftalima**
## Objective and Motivation
-  RESTful endpoint to server the React Frontend

## Serializers
* <code>UserSerializer</code>
* <code>BattleSerializer</code>
* <code>CreateBattleSerializer</code>

### BattleListView
-  `ListAPIView`
### CreateBattleView
- `CreateAPIView`

### BattleListTests
| Test     | Description  | Assert     |
| :---        |    :----   |   :--- |  
| <code>test_return_200_when_accessing_endpoint </code>  | to-do |  <code>assertResponse200 </code>   |
| <code>test_return_queryset </code>  | to-do |  <code>assertTrue </code> ,   <code>assertEqual </code> |
| <code>test_return_empty_queryset </code>  | to-do |  <code>assertFalse </code>   |

### CODE REVIEW

File: backend/battles/api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        fields = ("id", "email")
+
+
+class BattleSerializer(serializers.ModelSerializer):
+    creator = UserSerializer()
+    opponent = UserSerializer()
+
+    class Meta:
+        model = Battle
+        fields = ("id", "creator", "opponent", "winner", "created_at")
```
Comment (rsarai) [inline backend/battles/api/serializers.py:20]: Shouldn't the winner be defined above as well

File: backend/battles/api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class CreateBattleSerializer(serializers.ModelSerializer):
+    opponent = serializers.CharField(style={"base_template": "textarea.html"})
```
Comment (rsarai) [inline backend/battles/api/serializers.py:24]: Why do you need this: `style={"base_template": "textarea.html"}`?

File: backend/battles/forms.py [hunk 3]
```diff
@@ -44,10 +44,7 @@ def clean_opponent(self):
                 raise forms.ValidationError("ERROR: You can't challenge yourself.")
         except User.DoesNotExist:
             self.is_guest = True
-            opponent = User.objects.create(email=opponent_email)
-            random_password = get_random_string(length=64)
-            opponent.set_password(random_password)
-            opponent.save()
+            opponent = create_guest_opponent(opponent_email)
```
Comment (rsarai) [inline backend/battles/forms.py:47]: Nice!

File: backend/battles/api/views.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    serializer_class = BattleSerializer
+
+    def get_queryset(self):
+        queryset = Battle.objects.filter(
+            Q(creator=self.request.user) | Q(opponent=self.request.user)
+        ).order_by("-id")
+        return queryset
+
+
+class CreateBattleView(generics.CreateAPIView):
```
Comment (rsarai) [inline backend/battles/api/views.py:19]: Shouldn't you send the email here: https://github.com/naftalima/pokebattle/blob/main/backend/battles/views.py#L32 ?

File: backend/battles/api/urls.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path
+
+from battles.api.views import BattleListView, CreateBattleView
+
+
+urlpatterns = [
+    path("battle/", BattleListView.as_view(), name="battle_list"),
+    path("battle/new/", CreateBattleView.as_view(), name="battle_create"),
```
Comment (rsarai) [inline backend/battles/api/urls.py:8]: are the `name` attributes unique?

**Review by rsarai — CHANGES_REQUESTED** [review]
Looking good overall, but please remove the empty files: `backend/battles/serializers.py`, `backend/battles/endpoints.py`, and `backend/battles/drf_urls.py`

File: backend/battles/api/serializers.py (hunk 2 above)
Comment (naftalima) [reply backend/battles/api/serializers.py:24]: I don't need to, but I didn't know that - I was using the django-rest-framework tutorial as a base. You can see it here: https://www.django-rest-framework.org/tutorial/1-serialization/#creating-a-serializer-class

removed

File: backend/battles/api/views.py (hunk 4 above)
Comment (naftalima) [reply backend/battles/api/views.py:19]: I added sending emails, but in this PR: https://github.com/naftalima/pokebattle/pull/55

File: backend/battles/api/urls.py (hunk 5 above)
Comment (naftalima) [reply backend/battles/api/urls.py:8]: yes

File: backend/battles/api/serializers.py (hunk 1 above)
Comment (naftalima) [reply backend/battles/api/serializers.py:20]: I don't quite understand. Is this a question? 
It's working like this, what should I change?

**Review by rsarai — APPROVED** [review]

**Pull request merged**